src/lib/product/ProductExport.ts

```diff
@@ -410,10 +419,20 @@ export class ProductExport {
 
     private wrapProductCharges(productCharges:Array<any>) {
         productCharges.forEach((charge) => {
-            const product = this.products.find(e => e.record['enxCPQ__TECH_External_Id__c'] === charge['enxCPQ__Root_Product__r']['enxCPQ__TECH_External_Id__c']);
-            if (product !== undefined) {
-                product.charges.push(charge);
-            }
+          let product;
+          let resource;
```

Comment:
W JSie i TSie lepiej jest tworzyć stałe niemutowalne, niż mutować istniejące:
https://www.geeksforgeeks.org/why-is-immutability-so-important-in-javascript/
Tu w zasadzie to niewiele zmieni, ale warto uważać z let-ami i var-ami.